lib/compilers.js

```diff
@@ -106,22 +106,32 @@ module.exports = function(extglob) {
       return this.mapVisit(node.nodes);
     })
     .set('paren.open', function(node) {
+      var capture = this.options.capture ? '(' : '';
+
       switch (node.parent.prefix) {
         case '!':
         case '^':
-          return this.emit('(?:(?!(?:', node);
+          return this.emit(capture + '(?:(?!(?:', node);
         case '*':
         case '+':
         case '?':
         case '@':
-          return this.emit('(', node);
+          return this.emit(capture + '(?:', node);
         default: {
-          var val = (this.options.bash === true ? '\\' : '') + node.val;
+          var val = node.val;
+          if (this.options.bash === true) {
+            val = '\\' + val;
+          } else if (!this.options.capture && val === '(' && node.parent.rest[0] !== '?') {
```

Comment:
Not sure exactly what to do here. This is to work around [this test](https://github.com/micromatch/extglob/blob/master/test/test.js#L67) (`!((?:b/a))`) which already have the characters that make a non-capturing group. Not sure if they are supposed to be escaped, but they aren't currently so adding them again here breaks the regex.

Reply:
I'll take a look. thanks